COLI-314: add a separate Orchard 1 ruleset to adjust independently of… (#20)
Lombiq/.NET-Analyzers: Docs/AddingAnalyzers.md

</PropertyGroup>
```

> ## How to add the analyzers to individual .NET Framework projects (not solutions)

Reviewer: Add a note that this is only applicable for the legacy csprojs.

1. Same as above - add the .NET-Analyzers repository as a submodule to your repository.
2. Create a *Directory.Build.props* file in every project folder you want to target, next to the *.csproj* file with the following content (import the *`NetFx.Build.props`* file instead of *Build.props*):
   ```xml
   <Project>

>      <Import Project="tools/Lombiq.Analyzers/NetFx.Build.props"/>

Reviewer: ```suggestion
     <Import Project="../../../tools/Lombiq.Analyzers/NetFx.Build.props"/>
```

While it might not be exactly this it won't be just *tools* for sure.

BTW This is using forward slashes but the files in COLI contain "..\..\..\tools\Lombiq.Analyzers\NetFx.Build.props". Which one is good?

Author: Both are OK - here I used the same syntax as further up in the same file.

Do you think it's necessary to adjust the files in COLI with regard to this documentation?

Reviewer: No, just the doc is OK.

   </Project>
   ```
3. The *NetFx.Build.props* will copy this project's *.editorconfig* file into every project folder that you've created a *Directory.Build.props* in, so you might want to gitignore those:
    ```
    .editorconfig
    ```
    

> This will use the analyzer configuration suitable for Orchard 1 projects. If you want to use the this in a non-Orchard .NET Framework app then use the *General.ruleset* file as described above.

Reviewer: ```suggestion
This will use the analyzer configuration suitable for Orchard 1 projects. If you want to use this in a non-Orchard .NET Framework app then use the *General.ruleset* file as described above.
```

        <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
    </PropertyGroup>
    ```

> 3. This project can be used with .NET Framework via *NetFx.Build.props* but the including project needs to use `PackageReference`s. You can convert your `packages.config` based project to `PackageReferences` using [this walkthrough](https://docs.microsoft.com/en-us/nuget/consume-packages/migrate-packages-config-to-package-reference). .NET Core and .NET 5 or later projects need to use the new SDK-style csproj format. You [can convert projects manually](https://docs.microsoft.com/en-us/dotnet/core/porting/#per-project-steps) or automatically with the [try-convert](https://github.com/dotnet/try-convert) utility. You'll also need to manually remove any leftover `packages.config` files and adjust `NuGet.config` files.

Reviewer: ```suggestion
3. .NET compatibility:
    - .NET Core and .NET 5 or later projects can use this without anything special.
    - .NET Framework projects:
      - Projects using the new SDK-style csproj format don't need anything special. You [can convert projects manually](https://docs.microsoft.com/en-us/dotnet/core/porting/#per-project-steps) from the legacy csproj format or automatically with the [try-convert](https://github.com/dotnet/try-convert) utility. You'll also need to manually remove any leftover *packages.config* files and adjust *NuGet.config* files.
      - Projects using the legacy csproj format should use *NetFx.Build.props* as described above but the including project needs to use `PackageReference`s. You can convert your *packages.config*-based project to `PackageReference`s using [this walkthrough](https://docs.microsoft.com/en-us/nuget/consume-packages/migrate-packages-config-to-package-reference).  
```
